pkg/odo/cli/dev/dev.go

```diff
@@ -214,11 +168,11 @@ func (o *DevOptions) HandleSignal() error {
 	select {}
 }
 
-func (o *DevOptions) Cleanup(commandError error) {
+func (o *DevOptions) Cleanup(ctx context.Context, commandError error) {
 	if commandError != nil {
-		devFileObj := o.Context.EnvSpecificInfo.GetDevfileObj()
-		componentName := o.GetComponentName()
-		_ = o.clientset.WatchClient.CleanupDevResources(devFileObj, componentName, log.GetStdout())
+		devFileObj := odocontext.GetDevfileObj(ctx)
+		componentName := odocontext.GetComponentName(ctx)
+		_ = o.clientset.WatchClient.CleanupDevResources(ctx, *devFileObj, componentName, log.GetStdout())
```

Comment:
Would it make sense to change the `watch` interface and implementation, so clients could just pass a Context to `CleanupDevResources`, e.g.:
```go
_ = o.clientset.WatchClient.CleanupDevResources(ctx, log.GetStdout())
```

Reply:
There are many places where this kind of refactoring needs to be done, I would prefer to do this in specific PRs, to not add more to this one.

We have to take care in which packages we want to propagate this context (for example, the odocontext.Get* functions should not be used in low-level packages like kclient, exec, sync, etc but only on odo-specific packages.

Reply:
Makes sense - thanks.